app/src/main/java/com/cornellappdev/android/eatery/model/DiningHallModel.java

```diff
@@ -0,0 +1,290 @@
+import org.threeten.bp.LocalDate;
+import org.threeten.bp.LocalDateTime;
+import org.threeten.bp.LocalTime;
+import org.threeten.bp.ZoneId;
+import org.threeten.bp.ZonedDateTime;
+import org.threeten.bp.format.DateTimeFormatter;
+
+public class DiningHallModel extends EateryModel {
+
+  private Map<Integer, List<MealModel>> mWeeklyMenu;
+
+  public DiningHallModel() {
+    this.mWeeklyMenu = new HashMap<>();
+  }
+
+  public List<MealModel> getMenuForDay(DayOfWeek day) {
+    List<MealModel> val = mWeeklyMenu.get(day.getValue());
+
+    if (val != null) {
+      return val;
+    }
+
+    return Collections.emptyList();
+  }
+
+  public Map<DayOfWeek, List<MealModel>> getWeeklyMenu() {
+    Map<DayOfWeek, List<MealModel>> mapping = new HashMap<>();
+
+    for (Map.Entry<Integer, List<MealModel>> entry : mWeeklyMenu.entrySet()) {
+      mapping.put(DayOfWeek.of(entry.getKey()), entry.getValue());
+    }
+
+    for (DayOfWeek dayOfWeek : DayOfWeek.values()) {
+      if (!mapping.containsKey(dayOfWeek)) {
+        mapping.put(dayOfWeek, Collections.emptyList());
+      }
+    }
+
+    return mapping;
+  }
+
+  private void setMenuForDay(DayOfWeek day, List<MealModel> mWeeklyMenu) {
+    this.mWeeklyMenu.put(day.getValue(), mWeeklyMenu);
+  }
+
+  public List<String> getMealItems() {
+    List<String> items = new ArrayList<>();
+    Status status = getCurrentStatus();
+
+    if (status.isOpen()) {
+      LocalDateTime now = LocalDateTime.now();
+
+      for (Map.Entry<DayOfWeek, List<MealModel>> day : getWeeklyMenu().entrySet()) {
+        for (MealModel meal : day.getValue()) {
+          LocalDateTime startTime = meal.getStart(), endTime = meal.getEnd();
+
+          if (now.isAfter(startTime) && now.isBefore(endTime)) {
+            MealMenuModel menu = meal.getMenu();
+            items.addAll(menu.getAllItems());
+          }
+        }
+      }
+    }
+
+    return items;
+  }
+
+  @Override
+  public ZonedDateTime getNextOpening() {
+    ZonedDateTime time = ZonedDateTime.now();
+
+    for (Map.Entry<DayOfWeek, List<MealModel>> day : getWeeklyMenu().entrySet()) {
+      for (MealModel meal : day.getValue()) {
+        ZoneId cornell = TimeUtil.getInstance().getCornellTimeZone();
+
+        ZonedDateTime startTime = meal.getStart().atZone(cornell);
+
+        if (startTime.isAfter(time)) {
+          return startTime;
+        }
+      }
+    }
+    return null;
+  }
+
+  public String stringTo() {
+    String info = "Name/mNickName: " + mName + "/" + mNickName;
+    String locationString = "Location: " + ", Area: " + mArea;
+    String payMethodsString = "Pay Methods: " + mPayMethods.toString();
+    StringBuilder menuString = new StringBuilder();
+
+    for (Map.Entry<DayOfWeek, List<MealModel>> day : getWeeklyMenu().entrySet()) {
+      for (MealModel meal : day.getValue()) {
+        menuString.append(meal.stringTo());
+      }
+    }
+
+    return info + "\n" + locationString + "\n" + payMethodsString + "\n" + "Menu" + "\n"
```

Comment:
Same comment here

Reply:
Also a relic, will fix all stringTo issues in a later merge.